packages/gestalt/src/Status.js

```diff
@@ -0,0 +1,94 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import Icon from './Icon.js';
+import Flex from './Flex.js';
+import Text from './Text.js';
+
+export type StatusColor = 'gray' | 'green' | 'orange' | 'red';
+
+type Props = {|
+  type?: 'unstarted' | 'inProgress' | 'halted' | 'ok' | 'problem' | 'canceled' | 'warning',
+  title?: string,
+  subtext?: string,
+|};
+
+export default function Status(props: Props): Node {
+  const { type = 'unstarted', title, subtext } = props;
+
+  let color;
+  let icon;
+
+  /*
+   * QUESTION: Are switch / case conditionals OK? I've had folks balk at their use in the past.
+   */
+  switch (type) {
```

Comment:
This works, but I think a couple map might be a cleaner solution here:
```js
const ICON_COLOR_MAP = {
  unstarted: {
    icon: 'workflow-status-unstarted',
    color: 'darkGray`',
  },
  inProgress: {
    icon: 'workflow-status-in-progress',
    color: 'green',
  },
  ...
};

function Status({ type = 'unstarted', title, subtext }: Props) {
  const { icon, color } = ICON_COLOR_MAP[type];

  ...
}
```

Reply:
This is _exactly_ what I was hoping to learn - so much better. Thank you!